src/pages/general/login/actions/loginRequest.ts

```diff
@@ -0,0 +1,25 @@
+import { hashHistory } from 'react-router';
+import { LoginApi } from './../../../../rest-api/login';
+import { LoginCredentials } from './../../../../model/loginCredentials';
+import { loginActionEnums } from './../../../../common/actionEnums/login';
+import { LoginResponse } from './../../../../model/loginResponse';
+
+export const loginRequestStarted = (loginCredentials : LoginCredentials) => {
+  return function(dispatcher) {
+    const promise = LoginApi.login(loginCredentials);
+
+    promise.then( data => {
+      dispatcher(loginRequestCompleted(data));
+
+      if(data.succeded === true) {
+        hashHistory.push('/students');
+      }
```

Comment:
Rather let's go for the action approach now (to avoid refactor twice the unit tests):

- Let's create an action: NavigateToHomeBasedOnRole it will accept as parameter the role of the logged in user.

- Depending on the role it will navigate to one of the home pages (student, trainer, admin).

- Tests:
   - On the login action side, just check that this action is called or dispatched (on login successful).

   - On the action itself (rol navigation), check that on each role is fired the right navigation.